openeogeotrellis/deploy/batch_job.py

```diff
@@ -120,6 +125,8 @@ def extract_result_metadata(tracer: DryRunDataTracer) -> dict:
             logger.warning("Multiple aggregate_spatial geometries: {c}".format(c=len(aggregate_spatial_geometries)))
         agg_geometry = aggregate_spatial_geometries[0]
         if isinstance(agg_geometry, BaseGeometry):
+            # We only allow EPSG:4326 for the BaseGeometry case.
```

Comment:
Is this because the DryRunDataTracer will always return BaseGeometries in EPSG:4326?

Reply:
No, this was more of a pragmatic decision to keep things simple and prevent complicated cases with different CRSs.

The BaseGeometries are objects from Shapely, but Shapely doesn't really support CRS transformations, meaning all operations with two geometries expect that the geometries have the same CRS.
The Shapely docs don't seem to say much about it though:
ref: https://shapely.readthedocs.io/en/stable/manual.html#coordinate-systems

While it would be technically possible to handle BaseGeometries with different CRSs , it looks like there could be a lot of issues to take into account, and there might not be much demand to provide this capability.
So this limitation is more of a costs vs benefits decision to limit the allowed inputs so we can keep it simple.